src/lib/bootloader/stage1.rb

```diff
@@ -158,20 +160,20 @@ def available_locations
     end
 
     def can_use_boot?
-      tm = Yast::Storage.GetTargetMap
       partition = Yast::BootStorage.BootPartitionDevice
 
-      part = Yast::Storage.GetPartition(tm, partition)
+      parts = partitions.with(name: partition)
```

Comment:
well, maybe in the end `Yast::BootStorage.BootPartitionDevice` can hold object instead of string, just idea, even less then NP :)